currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -264,7 +270,14 @@ def _scheduled_update(self):
                     else (provider.next_run - provider._get_next_run_period())
                 )
                 date_to = provider.next_run
-                provider._update(date_from, date_to, newest_only=True)
+                if (date_to != fields.Date.today()) or (
+                    date_to == fields.Date.today()
+                    and (
+                        not provider._get_close_time()
+                        or datetime.now().hour >= provider._get_close_time()
```

Comment:
Is it intended to operate in UTC?

Reply:
Yes, it is important to do the hour comparison in UTC in this condition.

Reply:
```
>>> datetime.now()
datetime.datetime(2022, 3, 8, 21, 50, 30, 339383)
>>> datetime.utcnow()
datetime.datetime(2022, 3, 8, 20, 52, 47, 902066)
```
Here, the comparison is made using the LOCAL time of the odoo server, which is definitely not a good idea. _get_close_time() should also be renamed _get_utc_close_time() to make things clear.
The current code is really a mess, causing more bugs than solutions :(

Reply:
validation in utc is not a bug, it is meant to be done like this (discussion here: https://github.com/OCA/currency/pull/85 ). In any case, if you think the code can be improved, you can do it in a pull request with changes that you consider.